### DISCUSSION TITLE
How can I set an icon and a link to a file attachment (embedded file)

### DISCUSSION THREAD

**Discussion by FernandoJCabral** [pymupdf/PyMuPDF#1114] [Looking for help]
Embedding a pdf file into another pdf file is working fine. What a need next is a way to insert an icon that, once clicked, takes the reader to that attachment. Don't get me wrong: when reading the fine on evince or okular, they both inform the reader that there are attachments. This works fine. But, sometimes a need to put a reference as a hyperlink right where the attachment should be consulted. I did it with annotation, but I must do it as an embeded attachment. 

This works fine:

`file[0].addFileAnnot((0,0), embutido, "Anotação", ufilename=None, desc=None, icon="PushPin")`

This works, but I can't inser an hyperlink and an icon:

`file.embeddedFileAdd("Embedded.pdf", embedded)`

Any hints, please?

**Comment by JorjMcKie** [comment]
> This works, but I can't inser an hyperlink and an icon: ...

This is not supported yet, unfortunately. This type of hyperlink has - as per PDF definition - a link kind `GoToE` ("go to embedded").
If it were supported, it would and behave like any other hyperlink: not with an icon but with some clickable hot area.

I am accepting this as an enhancement request.

**Reply by MercyTrek** [reply]
Hi!
I was looking for something similar. Basically, add a hyperlink to text, clicking which, you go to the embedded PDF. I am able to embed the PDF but have no idea how to create a link to this embedded PDF. As it has been quite some time since this question was asked, I wanted to know if there is an update regarding the possibility of this being doable now?

**Reply by JorjMcKie** [reply]
This is still not supported.
